test/integration/serviceaccount/external_jwt_signer_test.go

```diff
@@ -245,6 +294,34 @@ func TestExternalJWTSigningAndAuth(t *testing.T) {
 			} else if tokenReviewResult.Status.Authenticated && !tc.shouldPassAuth {
 				t.Fatal("Expected Authentication to fail")
 			}
+
+			// Perform SSAR
```

Comment:
I actually think we should do a `SelfSubjectReview` which returns the authenticated user, and double-check 1) the request is allowed and 2) the returned user matches our expected service account username. You wouldn't need to create the pod read role/binding to accomplish that